src/cudamatrix/cu-kernels.cu

```diff
@@ -28,6 +28,16 @@
 #include <math_constants.h>
 #include "cudamatrix/cu-kernels-ansi.h"
 
+
+
+inline __device__ static float max_generic(float a, float b) {
+  return fmaxf(a, b);
```

Comment:
Math functions such as `fmax()` are already overloaded for both `double` and `float` as documented here. http://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH.html#group__CUDA__MATH

It seems that `max()` has been overloaded for all possible types in this header but this is not documented.
/usr/local/cuda/include/math_functions.h